File: sliding-window/3-Longest-Substring-Without-Repeating-Characters.go
func lengthOfLongestSubstring(s string) int {
    repeatCharacter := make(map[rune]bool)
    left := 0
    ans := 0
    for right, c := range s {
        for repeatCharacter[c] {
            delete(repeatCharacter, rune(s[left]))
            left++
        }
        repeatCharacter[c] = true
        ans = max(ans, right - left + 1)
    }
    return ans
}

func max(a int, b int) int {
    if a > b {
        return a
    }
    return b
}

File: reference/sliding-window/3-Longest-Substring-Without-Repeating-Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        duplicate_character = set()
        left = 0
        ans = 0
        for right in range(len(s)):
            while s[right] in duplicate_character:
                duplicate_character.remove(s[left])
                left += 1
            duplicate_character.add(s[right])
            ans = max(ans, right - left + 1)
        return ans